src/benchmark_runner.cc

```diff
@@ -148,10 +199,12 @@ class BenchmarkRunner {
       run_results.file_report_aggregates_only =
           (b.aggregation_report_mode & internal::ARM_FileReportAggregatesOnly);
     }
+  }
 
+  void do_work() {  // DO NOT CALL ME FROM THE CONSTRUCTOR!!!
```

Comment:
this is a sign of fragility. is there an alternative design that doesn't have this limitation?

Reply:
I can squash it into `get_results()`. You wouldn't be able to return data from ctor..
If we really don't like virtual functions here, i guess i could _try_ CRTP.

Reply:
CRTP it is.